server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java

```diff
@@ -655,7 +655,7 @@ private void dequeueMails(int times) {
                 .flatMap(mailQueueItem -> Mono.fromCallable(() -> {
                     mailQueueItem.done(true);
                     return mailQueueItem;
-                }).subscribeOn(Schedulers.elastic()))
+                }).subscribeOn(Schedulers.newBoundedElastic(2, 20, "DeQueueSchedulerTest")))
```

Comment:
Ah good spot, that got forgotten during the migration elastic -> bounderElastic.

I think though that we have a wrapper for that though? `ReactorUtils.BLOCKING_CALL_WRAPPER`?

Reply:
Ah no it's a test, forget that :)

Reply:
Q: Can you explain the rationale behind this fix? This looks weird to me cause `elastic` is supposed to be even more flexible than `BoundedElastic`.

Reply:
> Q: Can you explain the rationale behind this fix? This looks weird to me cause `elastic` is supposed to be even more flexible than `BoundedElastic`.

It looks like the switch threads are a reason for `channel is already closed due to clean channel shutdown` 
I tried to send `ack` messages NOT in the reactive pipeline, We will get immediately an exception. 
The current code, we can re-produce it by repeat `org.apache.james.queue.rabbitmq.RabbitMQMailQueueTest.MailQueueSizeMetricsEnabled#browseAndDequeueShouldCombineWellWhenDifferentSlices` until failure.